clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -338,17 +338,20 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
     if (type_is_instr(memref.instr.type) ||
         memref.instr.type == TRACE_TYPE_PREFETCH_INSTR ||
         memref.instr.type == TRACE_TYPE_INSTR_NO_FETCH) {
-        // We do not bother to support legacy traces without encodings.
-        bool has_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
-        addr_t trace_pc = shard->prev_instr_.instr.addr;
-        instr_t cur_instr_decoded;
-        instr_init(GLOBAL_DCONTEXT, &cur_instr_decoded);
-        const app_pc decode_pc = const_cast<app_pc>(shard->prev_instr_.instr.encoding);
-        const app_pc next_pc =
-            decode_from_copy(GLOBAL_DCONTEXT, decode_pc,
-                             reinterpret_cast<app_pc>(trace_pc), &cur_instr_decoded);
-        if (next_pc == nullptr) {
-            instr_free(GLOBAL_DCONTEXT, &cur_instr_decoded);
+        bool expect_encoding = TESTANY(OFFLINE_FILE_TYPE_ENCODINGS, shard->file_type_);
+        addr_t trace_pc = memref.instr.addr;
+        std::unique_ptr<instr_t> cur_instr_decoded(new instr_t());
+        app_pc next_pc = nullptr;
```

Comment:
Move the next_pc declaration to the narrower if-block scope below.

Reply:
I see that the next_pc declaration wasn't moved. Was there any concern in moving it to the if-block?

Please see the section on code reviews here: https://dynamorio.org/page_code_reviews.html#autotoc_md114. It's usually good to have an explicit reply to each comment thread in addition to marking them resolved.

Reply:
Initially was using `next_pc != nullptr` check down below and it would've gone out of scope but now that we don't use that check, moving it inside the if-block as suggested.